implement comments-export for single application

sc15zs opened the pull request:
Ticket:
https://mhclgdigital.atlassian.net.mcas.ms/browse/FLS-1376

Description:
We have previously added the functionality to export all comments of all applications. 
This PR extends this functionality, to allow users to export all comments of a single application.
The link has been added to the application tasklist page on ASSESS:
<img width="1690" height="530" alt="Screenshot 2025-07-21" src="https://github.com/user-attachments/assets/66b35331-47d6-490f-94b8-a7ecaacf0032" />

Example export file:
[comments_export_CHAM_REG_Application_ID_423f60e9-e606-4edb-aa9f-e5a22a436257_21-07-2025.xlsx](https://github.com/user-attachments/files/21344952/comments_export_CHAM_REG_Application_ID_423f60e9-e606-4edb-aa9f-e5a22a436257_21-07-2025.xlsx)

How to test:
In ASSESS, go to any application and add a few comments. Both application-level comments and sub-criteria comments. 
Click the "Export assessment comments" link, and a file should be generated, including all comments of the given application.

wjrm500 commented on pre_award/application_store/db/queries/comments/queries.py:
```diff
@@ -13,20 +13,22 @@
         .join(Account, Comment.user_id == cast(Account.id, String))
         .filter(*filters)
-        .order_by(Comment.application_id, Comment.sub_criteria_id)
+        .order_by(Comment.application_id, Comment.sub_criteria_id, Comment.date_created)
```
When I test locally, seems like comments aren't actually getting ordered by date_created within their subcriteria, see screenshot:

<img width="730" height="97" alt="Image" src="https://github.com/user-attachments/assets/6c338c3c-1c09-4da5-83b6-e716614190b7" />

wjrm500 reviewed (changes requested):
Looking really good - works well on test and code all makes sense.

Apart from the potential issue with comments not being ordered correctly by date_created in the export (see my comment), there are a couple of small things to flag which I discovered while testing, which to be honest fall outside the scope of this PR (as you are leveraging existing functionality), but I'll mention them anyway for future reference:

- If you edit a comment, all versions of that comment appear in the export, with no way of identifying which comment is the visible one that appears on the web page
- The field "Sub-criteria ID" in the export feels unnecessary esp. as we already show "Sub-criteria name"
- If an application has no comments, you get a completely blank spreadsheet, without even a header row - I'd have expected a header row even with no data

But in terms of things to action on this PR before I can approve, just the comment ordering thing - and also, please add a test/s. Thanks 🏂

sc15zs commented:
> > Looking really good - works well on test and code all makes sense.
> > Apart from the potential issue with comments not being ordered correctly by date_created in the export (see my comment), there are a couple of small things to flag which I discovered while testing, which to be honest fall outside the scope of this PR (as you are leveraging existing functionality), but I'll mention them anyway for future reference:
> > 
> > * If you edit a comment, all versions of that comment appear in the export, with no way of identifying which comment is the visible one that appears on the web page
> > * The field "Sub-criteria ID" in the export feels unnecessary esp. as we already show "Sub-criteria name"
> > * If an application has no comments, you get a completely blank spreadsheet, without even a header row - I'd have expected a header row even with no data
> > 
> > But in terms of things to action on this PR specifically before I can approve, just the comment ordering thing - and also, please add a test/s. Thanks 🏂
> 
> @wjrm500 Hi Will, Thanks for the thorough testing. Noticing issue with ordering is a really good catch. Now it should be sorted. The problem was that we have a separate Comment and a CommentsUpdate model. It is the CommentsUpdate which actually holds the _comment text_ added by a user. The returned queryset was sorted by application IDs, sub-criteria IDs, and finally by Comment created_by date. But in the generated file the CommentsUpdate's created_by date was added, which is different from the Comment created by date, if a comment has been updated. Now, the queryset is sorted by the CommentsUpdate's created_by date (after by application ID and sub-criteria ID), which is as it should be.
> 
> Regarding the other points you've raised:
> 
> 1. If you edit a comment, all versions are added to the export: that's the expected behaviour, because on the the webpage also all versions of an edited comment is shown. This is to allow users to see who made a comment and what changes have been made, I guess for audit purposes. It's true for both application-level and sub-criteria comments.
> 2. Removed the sub-criteria ID column, it is indeed unnecessary, when we also have the sub-criteria name in the file.
> 3. You're right about being weird not to have a heading row in the generated file, when there is no data to be added. Now it's changed that the heading row is added in these cases.
> 
> Regarding the unit tests: working on it now.

@wjrm500 
Hi Will,
Now I've added a unit test.
